src/main/java/org/I0Itec/zkclient/IZkConnection.java

```diff
@@ -38,6 +38,8 @@
     public String create(String path, byte[] data, List<ACL> acl, CreateMode mode) throws KeeperException, InterruptedException;
 
     public void delete(String path) throws InterruptedException, KeeperException;
+    
+    public void delete(String path, int version) throws InterruptedException, KeeperException;
```

Comment:
Can you just update the existing delete method to take the version? The other methods don't have 2 options for a versioned vs. non-versioned request. Instead the implementation just passes -1 when it doesn't care.

`version could be changed to `expectedVersion` is consistent with the rest of the code.